Card/src/main/java/com/paypal/android/card/CardAPI.kt

```diff
@@ -12,33 +14,80 @@ internal class CardAPI(
     private val api: API,
     private val requestFactory: CardAPIRequestFactory = CardAPIRequestFactory()
 ) {
-    suspend fun confirmPaymentSource(orderID: String, card: Card): ConfirmPaymentSourceResult {
+    suspend fun confirmPaymentSource(orderID: String, card: Card): CardResult {
         val apiRequest = requestFactory.createConfirmPaymentSourceRequest(orderID, card)
         val httpResponse = api.send(apiRequest)
-        return if (httpResponse.status == HTTP_OK) {
-            runCatching {
-                val json = PaymentsJSON(httpResponse.body)
-                val status = json.getString("status")
-                val id = json.getString("id")
-                ConfirmPaymentSourceResult(response = OrderData(id, OrderStatus.valueOf(status)))
-            }.recover {
-                ConfirmPaymentSourceResult(error = OrderError("PARSING_ERROR", "Error parsing json response."))
-            }.getOrNull()!!
-        } else {
-            val json = PaymentsJSON(httpResponse.body)
-            val name = json.getString("name")
-            val message = json.getString("message")
+        val correlationId = httpResponse.headers["Paypal-Debug-Id"]
+        if (URLUtil.isValidUrl(apiRequest.path).not()) {
+            CardResult.Error(
+                payPalSDKError = APIClientError.invalidUrlRequest,
+                correlationID = correlationId
+            )
+        }
+        val bodyResponse = httpResponse.body
+        if (bodyResponse.isNullOrBlank()) {
+            return CardResult.Error(
+                payPalSDKError = APIClientError.noResponseData,
```

Comment:
qq: should this be considered an error? Thinking of [204 No Content](https://httpstatusdogs.com/204-no-content) here.

Reply:
Do you know what is the respective error Class for this on Java? This is so we can map it on "Http" data class. 
Ps. nice pic from status dogs haha

Reply:
Ah nice. Yeah haha that's a good site. I see what you're saying though I forgot this was `CardAPI`. It makes perfect sense to return an error in that case.